lib/membrane/rtcp/transport_feedback_packet/twcc.ex

```diff
@@ -89,13 +102,107 @@ defmodule Membrane.RTCP.TransportFeedbackPacket.TWCC do
     # https://datatracker.ietf.org/doc/html/draft-holmer-rmcat-transport-wide-cc-extensions-01#section-3.1
     reference_time = div(reference_time, Time.milliseconds(64))
 
-    payload =
-      <<base_seq_num::16, packet_status_count::16, reference_time::24, feedback_packet_count::8>> <>
-        encode_packet_status(packet_status_chunks) <> encode_receive_deltas(scaled_receive_deltas)
+    encoded_header =
+      <<base_seq_num::16, packet_status_count::16, reference_time::24, feedback_packet_count::8>>
+
+    encoded_packet_status_chunks =
+      Enum.map_join(packet_status_chunks, &encode_packet_status_chunk/1)
+
+    encoded_receive_deltas = Enum.map_join(scaled_receive_deltas, &encode_receive_delta/1)
+
+    [encoded_header, encoded_packet_status_chunks, encoded_receive_deltas]
+    |> Enum.join()
+    |> maybe_add_padding()
+  end
+
+  defp parse_feedback(payload, packet_status_count) do
+    with {:ok, {encoded_receive_deltas, parsed_packet_status}} <-
+           parse_packet_status(payload, packet_status_count, []),
+         {:ok, receive_deltas} <-
+           parse_receive_deltas(encoded_receive_deltas, parsed_packet_status, []) do
+      {:ok, receive_deltas}
+    else
+      {:error, reason} -> {:error, reason}
```

Comment:
Do you need to explicitly match on `{:error, reason}`? You could just omit the `else`